src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/frequentparsers/AddFrequentExpenseCommandParser.java

```diff
@@ -0,0 +1,57 @@
+package ay2021s1_cs2103_w16_3.finesse.logic.parser.frequentparsers;
+
+import static ay2021s1_cs2103_w16_3.finesse.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static ay2021s1_cs2103_w16_3.finesse.logic.parser.CliSyntax.PREFIX_TITLE;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import ay2021s1_cs2103_w16_3.finesse.logic.commands.frequent.AddFrequentExpenseCommand;
+import ay2021s1_cs2103_w16_3.finesse.logic.parser.ArgumentMultimap;
+import ay2021s1_cs2103_w16_3.finesse.logic.parser.ArgumentTokenizer;
+import ay2021s1_cs2103_w16_3.finesse.logic.parser.Parser;
+import ay2021s1_cs2103_w16_3.finesse.logic.parser.ParserUtil;
+import ay2021s1_cs2103_w16_3.finesse.logic.parser.Prefix;
+import ay2021s1_cs2103_w16_3.finesse.logic.parser.exceptions.ParseException;
+import ay2021s1_cs2103_w16_3.finesse.model.category.Category;
+import ay2021s1_cs2103_w16_3.finesse.model.frequent.FrequentExpense;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Amount;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Title;
+
+
+
+public class AddFrequentExpenseCommandParser implements Parser<AddFrequentExpenseCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddFrequentExpenseCommand
+     * and returns an AddFrequentExpenseCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddFrequentExpenseCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_TITLE, PREFIX_AMOUNT, PREFIX_CATEGORY);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_TITLE, PREFIX_AMOUNT)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddFrequentExpenseCommand.MESSAGE_USAGE));
+        }
+
+        Title title = ParserUtil.parseTitle(argMultimap.getValue(PREFIX_TITLE).get());
+        Amount amount = ParserUtil.parseAmount(argMultimap.getValue(PREFIX_AMOUNT).get());
+        Set<Category> categoryList = ParserUtil.parseCategories(argMultimap.getAllValues(PREFIX_CATEGORY));
+
+        FrequentExpense frequentExpense = new FrequentExpense(title, amount, categoryList);
+        return new AddFrequentExpenseCommand(frequentExpense);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
+}
```

Comment:
This method is repeated in `AddExpenseCommandParser` and `AddIncomeCommandParser`. Perhaps in a future PR this method could be placed somewhere else, then imported into classes that require it?